core/modules/savers/git-service.js

```diff
@@ -17,26 +17,23 @@ var base64utf8 = require("$:/core/modules/utils/base64-utf8/base64-utf8.module.j
 /*
 Select the appropriate saver module and set it up
 */
-var GitHubSaver = function(wiki) {
+var GitServiceSaver = function(wiki) {
 	this.wiki = wiki;
 };
 
-GitHubSaver.prototype.save = function(text,method,callback) {
+GitServiceSaver.prototype.save = function(text,method,callback) {
 	var self = this,
-		username = this.wiki.getTiddlerText("$:/GitHub/Username"),
-		password = $tw.utils.getPassword("github"),
-		repo = this.wiki.getTiddlerText("$:/GitHub/Repo"),
-		path = this.wiki.getTiddlerText("$:/GitHub/Path"),
-		filename = this.wiki.getTiddlerText("$:/GitHub/Filename"),
-		branch = this.wiki.getTiddlerText("$:/GitHub/Branch") || "master",
-		endpoint = this.wiki.getTiddlerText("$:/GitHub/ServerURL") || "https://api.github.com",
-		headers = {
-			"Accept": "application/vnd.github.v3+json",
-			"Content-Type": "application/json;charset=UTF-8",
-			"Authorization": "Basic " + window.btoa(username + ":" + password)
-		};
+		titleOfSelectedService = this.wiki.getTiddlerText("$:/GitService/ServiceType"),
+		service = require(titleOfSelectedService),
```

Comment:
I think it's dangerous to use a user-supplied tiddler value as a the name of a module to be required. It might set up a security vulnerability if the user chose a malicious value.

It would be better for service provider modules to be a new module type, handling them something like the way that the `is` filter operator invokes the submodules that implement each specific type of check.

Reply:
If we break the code into separate savers, I think this problem will be resolved, BTW I've got your idea.